azext_edge/edge/providers/orchestration/base.py

```diff
@@ -331,6 +306,35 @@ def prepare_sp(cmd, deployment_name: str, **kwargs) -> ServicePrincipal:
     )
 
 
+def ensure_correct_access(cmd, sp_app_id: str, existing_resource_access: List[dict]):
```

Comment:
cool refactor

Reply:
got tired of how big that freaking thing is